plugins/ecue.js

```diff
@@ -41,219 +34,141 @@ module.exports.export = function exportEcue(fixtures, options) {
+function exportHandleMode(xmlFixture, mode) {
+  let viewPosCount = 1;
+  for (let dmxCount = 0; dmxCount < mode.channels.length; dmxCount++) {      
+    let channel = mode.channels[dmxCount];
 
-  // if this is a switching channel, just use the default channel
-  if (chKey in switchingChannels) {
-    const triggerChannel = fixture.availableChannels[switchingChannels[chKey].triggerChannel];
-    const defaultValue = triggerChannel.defaultValue;
+    // skip unused channels
+    if (mode.channels[dmxCount] instanceof NullChannel) {
+      continue;
+    }
 
-    chKey = triggerChannel.capabilities.find(
-      cap => cap.range[0] <= defaultValue && defaultValue <= cap.range[1]
-    ).switchChannels[chKey];
-  }
-  
-  let channel = fixture.availableChannels[chKey];
+    // ecue doesn't support switching channels, so we just use the default channel but change the name
+    let switchingChannelName = null;
+    if (channel instanceof SwitchingChannel) {
+      switchingChannelName = channel.name;
+      channel = channel.defaultChannel;
+    }
 
-  if (chKey in fineChannels) {
-    // use coarse channel's data
-    channel = fixture.availableChannels[fineChannels[chKey]];
+    let fineChannelKey = null;
+    if (channel instanceof FineChannel) {
+      if (channel.fineness === 1) {
+        // ignore this channel, we handle it together with its coarse channel
+        continue;
+      }
 
-    const coarseChannelIndex = mode.channels.indexOf(fineChannels[chKey]);
+      // just pretend its a single channel and use coarse channel's data
+      fineChannelKey = channel.key;
+      channel = channel.coarseChannel;
+    }
 
-    if (coarseChannelIndex !== -1) {
-      dmxByte0 = coarseChannelIndex;
-      dmxByte1 = dmxCount;
+    let dmxByte0 = dmxCount + 1;
+    let dmxByte1 = 0;
 
-      mode.channels[dmxByte0] = null;
-      mode.channels[dmxByte1] = null;
+    if (fineChannelKey !== null) {
+      dmxByte0 = mode.getChannelIndex(fineChannelKey) + 1;
     }
-    else {
-      // coarse and first fine channel were already handled -> just pretend it's a single channel
-      channel.name = (channel.name || fineChannels[chKey]) + ' fine^' + (channel.fineChannelAliases.indexOf(chKey) + 1);
-      channel = divideChannelDmxValues(channel, channel.fineChannelAliases.length - 1);
+    else if (channel.fineChannelAliases.length > 0) {
+      dmxByte1 = mode.getChannelIndex(channel.fineChannelAliases[0]) + 1;
     }
-  }
-  else if ('fineChannelAliases' in fixture.availableChannels[chKey]) {
-    const firstFineAlias = fixture.availableChannels[chKey].fineChannelAliases[0];
-    const firstFineChannelIndex = mode.channels.findIndex(chKey => {
-      // used in a switching channel
-      if (chKey in switchingChannels) {
-        return switchingChannels[chKey].defaultChannel === firstFineAlias;
-      }
 
-      // used directly
-      return chKey === firstFineAlias;
+    const channelFineness = channel.getFinenessInMode(mode) === 1 ? 1 : 0;
```

Comment:
In a 24bit channel, fineness would be 0, so the DMX values are scaled down to 0-255. Is this intended? In my understanding, a 24bit channel should be splitted into a 16bit channel and a generic fine channel, so the main channel should scale its DMX values down to 0-(256²-1).